dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -1,5 +1,6 @@
 class Api::V1::SectionsController < Api::V1::JsonApiController
   load_resource :section, find_by: :code, only: [:join, :leave]
+  load_and_authorize_resource :sections, only: [:membership]
```

Comment:
This is not needed. The existing `load_and_authorize_resource` on line 5 already does this, because it runs for all actions except `join` and `leave`. By default the resource name is the same as the controller (in this case `section`, pluralized for collections), so this already fills `@sections`.

Now that I'm looking in here, the existing `index` collection action is also doing unnecessary work. In that case too, `@sections` is already filled by the CanCan load resource, so [this](https://github.com/code-dot-org/code-dot-org/blob/3e71223a593003f906d69860be0a43e75ca7034f/dashboard/app/controllers/api/v1/sections_controller.rb#L20-L22) `current_user.sections` can be changed to `@sections` in a followup.

Reply:
you're right. i assumed line 2 would only give me access to `@section` (even though you told me i wouldn't need another resource definition 🙃)

Reply:
Well, line *2* actually does only supply `@section`, but that's because it only applies to those 2 (member) actions, and it finds the section by `code` rather than id. That's a special case. Someone always has to be different ;)

Line *5* is our general (all actions except join and leave) load resource, which will work for this new collection action too.

Reply:
that makes much more sense. thanks for bearing with me while i learn CanCan 😂